### PR TITLE
Update Instances to use generated code

### PR DESCRIPTION
**Pull request by bowei**
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/gce/gce_instances.go [hunk 1]
```diff
@@ -298,55 +294,46 @@ func (gce *GCECloud) GetAllCurrentZones() (sets.String, error) {
...
+// TODO: this should be removed from the cloud provider.
 func (gce *GCECloud) GetAllZonesFromCloudProvider() (sets.String, error) {
 	zones := sets.NewString()
-
 	for _, zone := range gce.managedZones {
-		mc := newInstancesMetricContext("list", zone)
-		// We only retrieve one page in each zone - we only care about existence
-		listCall := gce.service.Instances.List(gce.projectID, zone)
-
-		listCall = listCall.Fields("items(name)")
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_instances.go]: You're okay losing this optimization?

Comment (nicksardo) [reply pkg/cloudprovider/providers/gce/gce_instances.go]: WDYT about adding the limit param to `ListXXX`?

**Comment by saad-ali** [comment]
/approve no-issue

Will remove LGTM and let @nicksardo add it

File: pkg/cloudprovider/providers/gce/gce_instances.go (hunk 1 above)
Comment (nicksardo) [reply pkg/cloudprovider/providers/gce/gce_instances.go]: If this is only for e2e, then it's not worth the optimization. NM.

File: pkg/cloudprovider/providers/gce/gce_instances.go [hunk 2]
```diff
@@ -407,87 +393,70 @@ func (gce *GCECloud) AddAliasToInstance(nodeName types.NodeName, alias *net.IPNe
...
+			remaining--
 		}
-		if page >= maxPages {
-			glog.Errorf("getInstancesByNames exceeded maxPages=%d for Instances.List: truncating.", maxPages)
+	}
+
+	if remaining > 0 {
+		var failed []string
+		for k := range found {
+			failed = append(failed, k)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_instances.go]: Don't you need to check that the value is nil prior to adding it to the list?

File: pkg/cloudprovider/providers/gce/gce_instances.go [hunk 3]
```diff
@@ -407,87 +393,70 @@ func (gce *GCECloud) AddAliasToInstance(nodeName types.NodeName, alias *net.IPNe
...
+	instanceArray := make([]*gceInstance, 0, len(names))
+	for _, name := range names {
 		name = canonicalizeInstanceName(name)
-		instance := instances[name]
-		if instance == nil {
-			glog.Errorf("Failed to retrieve instance: %q", name)
-			return nil, cloudprovider.InstanceNotFound
-		}
-		instanceArray[i] = instances[name]
+		instanceArray = append(instanceArray, found[name])
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_instances.go]: Instead of touching `names` and `found` to create `instanceArray`, can we just iterate `found`?

**Comment by bowei** [comment]
updated

**Pull request merged**